### DISCUSSION TITLE
HA & MQTT in docker ... help  :'-(

### DISCUSSION THREAD

**Discussion by riccionese** [indomus/forum#1015] [HUB: Home Assistant]
ragazzi HELP !    ... non ci cavo gli zampetti !
sto crcando di protare tutto su docker
ho studiato abbastanza docker ed ho seguito le guide InDomus 
su una macchina ubuntu con doker ho dockerizzato 
- pihole
- unbound
- moquitto 
- home assistant

mi funziona tutto compreso HA e le sue integrazioni varie ... tranne home HA vs MQTT.
o meglio 
  - MQTT funziona --> con MQTT exploer da "da fuori" vedo tutto e posso pubblicare topics
  - HA si logga --> sempre con MQTT lo vedo "online"
  - il dispositivo con sui sto provando si registra --> e pubblicando il topic "toggle" succede quello che ci si aspetta

Ma in HA non risco ad avere lo swtch rilevato.
in configuration.yaml ho lo stesso bocco che ho per tutti gli altri devce shelly in MQTT nel HA core venv "in produzione" e cioè questo:
switch:
```
  - platform: mqtt
    name: "shelly1-new1"
    state_topic: "shellies/shelly1-new1/relay/0"
    command_topic: "shellies/shelly1-new1/relay/0/command"
    availability_topic: "shellies/shelly1-new1/online"
    qos: 1
    value_template: "{% if value == 'on' %} on {% else %} off {% endif %}"
    payload_on: "on"
    payload_off: "off"
    payload_available: "true"
    payload_not_available: "false"
    optimistic: false
```
ma niente da fare
mi sono anche collegato all'interno del container di HA (```docker exec -it home-assistant bash```) e con ```nmap``` (che per fortuna è già presente nell'immagine di HA) ho verificato che nel container di MQTT la porta sia in effetti ```open``` .... e lo è

cosa mi sto dimenticando / sbagliando ?
help ...

**Comment by riccionese** [comment]
aggiungo: da dentro HA, dalla funzione di "Ascolta un argomento" della integrazione MQTT, infatti vedo passare i comandi che lo shelly riceve/gestisce
![image](https://user-images.githubusercontent.com/62329292/140704917-725d1999-0470-4fce-9a40-01c2118bd2cc.png)
ma lo switch risulta sempre disabilitato.
A questo punto deve per forza essere qualche cosa nel settaggio dello switch ma non riesco a capire cosa

**Reply by riccionese** [reply]
anche pubblicando da dentro la config dell'integrazione MQTT sembra vada tutto bene 
![image](https://user-images.githubusercontent.com/62329292/140721805-6f0c0f58-dc74-4fe1-9ac5-8290b6d9612e.png)

ma ...
![image](https://user-images.githubusercontent.com/62329292/140721856-c69a826b-e0a4-4971-8687-b08af36792e7.png)

**Reply by snakuzzo** [reply]
ti sei ricordato dell'automazione di announce ?

**Reply by riccionese** [reply]
ovvio che no!!!! 
ma porc@#%$]#@!!$% !!!!!
indovina ? adesso funziona tutto 
ma ri-porc@#%$]#@!!$% !!!!!
(brutta roba la vecchiaia!)

Grazie, San Snakuzzo !

**Reply by snakuzzo** [reply]
:)
